contrib/githooks/pre-commit

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -e
+
+CMDS='git go gofmt goimports misspell'
+STAGED_GO_FILES=$(git diff --cached --name-only -- '*.go')
+
+f_echo_stderr() {
+  echo $@ >&2
+}
+
+f_exit_success() {
+  [ x"$@" != "x" ] && f_echo_stderr $@ || exit 0
+}
+trap f_exit_success EXIT
+
+f_check_cmds() {
+  for cmd in ${CMDS}; do
+    which ${cmd} &>/dev/null || f_exit_success "couldn't find ${cmd}, skipping"
+  done
+}
+
+f_check_cmds
+
+if [[ $STAGED_GO_FILES != "" ]]; then
+  f_echo_stderr "[pre-commit] fmt'ing staged files..."
+  for file in $STAGED_GO_FILES; do
+    if [[ $file =~ vendor/ ]] || [[ $file =~ client/lcd/statik/ ]] || [[ $file =~ tests/mocks/ ]]; then
```

Comment:
will this format *.pb.go files

Reply:
nvm on the /vendor

Reply:
https://github.com/cosmos/cosmos-sdk/pull/6041/commits/0f86b5456e72335bbbde426ff93253c192406360 - thanks, fixed